#!/usr/bin/python
PACKAGE = "wheel_leg_fsm"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Q_inertial_x", double_t, 0, "weight of inertial x", 3300, 1, 10000)
gen.add("Q_inertial_y", double_t, 0, "weight of inertial y", 3300, 1, 10000)
gen.add("Q_xv", double_t, 0, "weight of velocity in x axis", 385.0, 0.1, 500)
gen.add("Q_y", double_t, 0, "weight of y position", 3000, 1, 10000)
# gen.add("y_d_ratio", double_t, 0, "y_d_ratio", 1, 0, 1)
gen.add("Q_z", double_t, 0, "weight of z position", 400, 100, 1000)
# gen.add("z_d_ratio", double_t, 0, "z_d_ratio", 1, 0, 1)
gen.add("Q_roll", double_t, 0, "weight of roll", 300, 10, 600)
# gen.add("roll_d_ratio", double_t, 0, "roll_d_ratio", 1, 0, 1)
gen.add("Q_pitch", double_t, 0, "weight of pitch", 250, 10, 3500)
# gen.add("pitch_d_ratio", double_t, 0, "pitch_d_ratio", 1.0, 0, 1)
gen.add("Q_yaw", double_t, 0, "weight of yaw rate", 80, 10, 1000)
gen.add("Q_dx", double_t, 0, "weight of delta x", 8000, 100, 18200)
# gen.add("dx_d_ratio", double_t, 0, "weight of delta x", 1.0, 0.8, 3.0)




gen.add("R_N", double_t, 0, "weight of GRF in x axis", 0.5, 0.001, 10) # name type flag description default min max
gen.add("R_Q", double_t, 0, "weight of GRF in y axis", 0.3, 0.001, 10)
gen.add("R_P", double_t, 0, "weight of GRF in z axis", 0.01, 0.001, 1)
gen.add("R_tao", double_t, 0, "weight of GRT in y axis", 0.6, 0.01, 10)

exit(gen.generate(PACKAGE, "wheel_leg_fsm", "MPC"))
